### PR TITLE
Add support for sensitive fields

### PR DESCRIPTION
**Pull request by nicosp**
Add support for sensitive fields to allow tracking changes without exposing actual information.

### CODE REVIEW

File: src/Model/Behavior/AuditLogBehavior.php [hunk 1]
```diff
@@ -88,6 +89,26 @@ public function injectTracking(
         }
     }
 
+    /**
+     * Redacts sensitive fields from the array
+     *
+     * @param array<string,mixed> $fields Field
```
Comment (dereuromark) [inline src/Model/Behavior/AuditLogBehavior.php]:
```suggestion
     * @param array<string, mixed> $fields Field
```

File: src/Model/Behavior/AuditLogBehavior.php [hunk 2]
```diff
@@ -88,6 +89,26 @@ public function injectTracking(
         }
     }
 
+    /**
+     * Redacts sensitive fields from the array
+     *
+     * @param array<string, mixed> $fields Field
+     * @return void
+     */
+    private function redactArray(array &$fields): void
```
Comment (cnizzardini) [inline src/Model/Behavior/AuditLogBehavior.php:98]: Why are we passing by reference here and below?

**Comment by cnizzardini** [comment]
Good feature addition!

File: src/Model/Behavior/AuditLogBehavior.php (hunk 2 above)
Comment (nicosp) [reply src/Model/Behavior/AuditLogBehavior.php:98]: The idea is to change the array inline to avoid copies because extract is already a copy of the data.

**Comment by cnizzardini** [comment]
I am good with this PR @dereuromark

**Review by cnizzardini — APPROVED** [review]

**Pull request merged**